src/math.jl

```diff
@@ -16,8 +16,28 @@ Base.:/(l::Number, r::Quantity) = l * inv(r)
 Base.:/(l::Dimensions, r::Number) = Quantity(inv(r), l, true)
 Base.:/(l::Number, r::Dimensions) = Quantity(l, inv(r), true)
 
-Base.:+(l::Quantity, r::Quantity) = Quantity(l.value + r.value, l.dimensions, l.valid && r.valid && l.dimensions == r.dimensions)
-Base.:-(l::Quantity, r::Quantity) = Quantity(l.value - r.value, l.dimensions, l.valid && r.valid && l.dimensions == r.dimensions)
+function Base.:+(l::Quantity, r::Quantity)
+    if iszero(l)
```

Comment:
Wouldn't you want a DimensionError to still be raised even if `iszero` is true? If adding a 2D vector to a 3D vector, you would always want to raise an error, regardless of if one of them was zero, no?

It might not even be the case that `iszero` is defined for all possible `Quantity`, since the value of a `Quantity` might be an abstract object, rather than just a scalar real

Reply:
The use case is that there are places where a value is initialized with zero, and than += or -= are used to add other quantities to it.

This can be solved also e.g. in Sparspak. But IMHO it this belongs to the algebraic properties of Quantity. What is `zero` and what is `one` ?

Reply:
It looks like other unit packages also raise DimensionErrors even if the value is zero. e.g., astropy:

```python
from astropy import units as u

q = 10 * u.m # 10 meters
q2 = 10 * u.s # 10 seconds

q * 0 + q2
# ^ Raises UnitConversionsError
```

likewise with Unitful.jl

```julia
julia> using Unitful

julia> 0.0u"m" + 10.0u"s"
ERROR: DimensionError: 0.0 m and 10.0 s are not dimensionally compatible.
```

Reply:
I think "zero meters" + "one second" raising an error makes the most sense as a default? Thoughts?

Reply:
I had it running like this, so for sparspak I certainly could switch to this. But sparspak may not be the only place where this functionality may be needed.

Fundamentally this leads to the question how units are working algebraically. I'll try to think more about it to find a less ad-hoc approach. 

IMHO it would be good if DynamicQuantities could be used like Dual numbers, MultiFloats, Measurements and the like as a drop-in replacement for Float64. 
But I am aware that this may be not everyone's version for this package.

Reply:
I think even if there are a few additional packages that need this, it is better left out of the main package, because it will slow down every call to `+` with 3x as many comparisons on the dimension objects.